packages/discovery-provider/src/queries/get_comments.py

```diff
@@ -306,9 +307,10 @@ def get_track_comments(args, track_id, current_user_id=None):
                 == None,  # Check if parent_comment_id is null
                 or_(
                     CommentReport.comment_id == None,
-                    current_user_id == None,
-                    CommentReport.user_id != current_user_id,
-                    CommentReport.user_id != artist_id,
+                    and_(
+                        CommentReport.user_id != current_user_id,
+                        CommentReport.user_id != artist_id,
+                    ),
```

Comment:
I'm confused by this addition, isn't the OR case here correct? Whats the reasoning for the AND